src/webgl/p5.Shader.js

```diff
@@ -0,0 +1,330 @@
+/**
+ * This module defines the p5.Shader class
+ * @module Lights, Camera
+ * @submodule Shaders
+ * @for p5
+ * @requires core
+ */
+
+var p5 = require('../core/core');
+
+/**
+ * Shader class for WEBGL Mode
+ * @class p5.Shader
+ * @constructor
+ * @param {renderer} [p5.Renderer.GL] an instance of p5.RendererGL that
+ * will provide the GL context for this new p5.Shader
+ * @param {vertSrc} [String] source code for the vertex shader (as a string)
+ * @param {fragSrc} [String] source code for the fragment shader (as a string)
+ */
+p5.Shader = function(renderer, vertSrc, fragSrc) {
+  // TODO: adapt this to not take ids, but rather,
+  // to take the source for a vertex and fragment shader
+  // to enable custom shaders at some later date
+  this._renderer = renderer;
+  this._vertSrc = vertSrc;
+  this._fragSrc = fragSrc;
+  this._vertShader = -1;
+  this._fragShader = -1;
+  this.glProgram = 0;
+  this.loadedAttributes = false;
+  this.attributes = {};
+  this.loadedUniforms = false;
+  this.uniforms = {};
+  this.bound = false;
```

Comment:
if these are public they should probably be documented with @property tags. maybe this is coming later? or should we discuss having two types of public methods/properties -- documented and undocumented? (I know this is currently the case in the rest of the library, but not necessarily by design heh, so it would be useful at this point I think to clarify and try to be more strict moving forward. what are your thoughts?)